File: OGLEngine/StaticGeometryModel.m
//
//  StaticGeometryModel.m
//  jj
//
//  Created by Maciej Chmielewski on 14.02.2016.
//  Copyright © 2016 AppUnite. All rights reserved.
//

#import "StaticGeometryModel.h"

@interface StaticGeometryModel ()

@property (nonatomic, assign) GLKMatrix4 staticModelMatrix;

@end

@implementation StaticGeometryModel

- (instancetype)initWithModelMatrix:(GLKMatrix4)modelMatrix {
    self = [super init];
    if (self) {
        self.staticModelMatrix = modelMatrix;
    }
    return self;
}

- (GLKMatrix4)modelMatrix {
    return self.staticModelMatrix;
}

@end

File: OGLEngine/StaticGeometryModel.swift
//
//  StaticGeometryModel.swift
//  OGLEngine
//
//  Created by Maciej Chmielewski on 21.03.2016.
//  Copyright © 2016 MaciejCh. All rights reserved.
//

import Foundation
import GLKit

class StaticGeometryModel: GeometryModel {
    
    var staticModelMatrix: GLKMatrix4?
    
    override func modelMatrix() -> GLKMatrix4 {
        if (self.staticModelMatrix == nil) {
            self.staticModelMatrix = super.modelMatrix()
        }
        return self.staticModelMatrix!
    }
}